cpp/ql/lib/ext/Boost.Asio.model.yml

```diff
@@ -0,0 +1,26 @@
+extensions:
+  # partial model of the Boost::Asio network library
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: sourceModel
+    data: # namespace, type, subtypes, name, signature, ext, output, kind, provenance
+      - ["boost::asio", "", False, "read", "", "", "Argument[1*]", "remote", "manual"]
```

Comment:
I think we only support _prefixing_ dereferences, no? I certainly couldn't find anything for this in [decodePosition](https://github.com/github/codeql/blob/78552b8bcaf74fd17a6d49784220341d7fe25354/cpp/ql/lib/semmle/code/cpp/dataflow/internal/FlowSummaryImpl.qll#L56) 😅

Reply:
Yeah, that might be a typo.  Surprised the test is passing.  I'll look into it...

Reply:
Fixed (the test was passing because it doesn't have 100% coverage and the typos are only present in a few cases).